inst/templates/ad_adlb.R

```diff
@@ -0,0 +1,276 @@
+# Name: ADLB
+#
+# Label: Lab Analysis Dataset
+#
+# Input: adsl, lb
+library(admiral)
+library(admiral.test) # Contains example datasets from the CDISC pilot project
+library(dplyr)
+library(lubridate)
+library(stringr)
+
+# ---- Load source datasets ----
+
+# Use e.g. haven::read_sas to read in .sas7bdat, or other suitable functions
+# as needed and assign to the variables below.
+# For illustration purposes read in admiral test data
+
+data("lb")
+data("adsl")
+
+lb <- convert_blanks_to_na(lb)
+
+# The CDISC Pilot Data contains no SUPPVS data
+# If you have a SUPPLB then uncomment function below
+
+# lb <- derive_vars_suppqual(lb, supplb) %>%
+
+# ---- Lookup tables ----
+adlb_dis <- adlb %>%
+  distinct(LBTESTCD, LBSTRESU, LBTEST) %>%
+  arrange(LBTESTCD)
+
+# Assign PARAMCD, PARAM, and PARAMN
+param_lookup <- tibble::tribble(
```

Comment:
can we do something like this for the lookup table? param_lookup <- lb %>% 
  distinct(LBTEST, LBTESTCD, LBORRESU, LBCAT) %>% 
  mutate(PARAM = str_c(LBTEST, " (", LBORRESU, ")"),
         PARAMCD = LBTESTCD,
         PARCAT1 = str_to_sentence(LBCAT),
         PARAMN = seq(1:length(PARAMCD)))

Reply:
Maybe use dataset_vignette to view the lookup table?

Reply:
Depending on SDTM mapping rules, you may have missing LBSTRESU where AVAL is missing, so the PARAM for those obs woud be different to obs where AVAL and LBSTRESU populated, which would be wrong. (We should always use SI unit in LBSTRESU, not original unit LBORRESU.
I think its cleaner if study teams/companies supply their own look-up table, and just pass it on to {admiral}, the look-up table we show is just an example.